### DISCUSSION TITLE
Model Choice in Chat Missing

### DISCUSSION THREAD

**Discussion by gmoore-liv** [community/community#144316] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Model choice still isn't appearing in my copilot chat. Is this feature still being rolled out? I think my Copilot chat is up to date with updates.

![image](https://github.com/user-attachments/assets/1de33ab0-4f14-4de0-b6f4-3edf9bf4368d)

![image](https://github.com/user-attachments/assets/f3a8be4a-a6d3-426a-b05e-a8969090da81)

**Comment by huddmax** [comment, marked as answer]
It seems like this feature might still be in the process of a gradual rollout, so not all users will have immediate access to model selection in Copilot Chat. Here are a few things you can check to make sure you're up to date:

1. **Check the Version**: Ensure you’re using the latest version of Copilot Chat. The most recent update should be available on GitHub Desktop or through notifications within GitHub Copilot.

2. **Account Settings**: Sometimes features like this are enabled through specific account settings. Check your Copilot settings to see if there’s an option for model selection.

3. **Contact Support**: If this feature is essential for your work, you might want to reach out to GitHub Support directly to ask if there’s an estimated timeline for when it will be available to everyone.

**Reply by gmoore-liv** [reply]
Thanks, checking the latest version I don't have the latest version but I have auto update checked. So I think I just have to wait until the gradual rollout hits me.

**Comment by amitpatelx** [comment]
I have VS Code `1.99.3` and Github Copilot `1.311.0` and I can't see model selection dropdown

![Screenshot from 2025-04-30 11-37-54](https://github.com/user-attachments/assets/f0bcb137-ac9b-4de6-851e-641e6841408e)

**Reply by ArshAliAzam** [reply]
You're looking at the GitHub Copilot settings, but the model selection dropdown (used to pick between different AI models like GPT-3.5 or GPT-4) is not showing.

✅ Here's what to check:
1. Copilot Labs vs Copilot Chat
Make sure you are using GitHub Copilot Chat, not just Copilot Labs.
The model dropdown appears only in Copilot Chat (inside the chat panel, near the top).

2. Use the “Change Copilot Completion Model” command
Try this:

Press Ctrl + Shift + P (or Cmd + Shift + P on Mac) to open the Command Palette.

Type:
"Change Copilot Completion Model"

If this option shows up, click it.

You should then see the list of available models (like gpt-3.5-turbo, gpt-4, etc.)

If the command doesn't appear, it means you don’t have access to model selection yet (see point 4 below).

3. Check if you're signed in correctly
Make sure:

You're signed into GitHub inside VS Code.

You have an active Copilot subscription (especially Copilot Chat Beta).

4. Model selection is in limited rollout
GitHub is gradually rolling out model selection. If you don't see the dropdown or command, it likely means:

❌ You don't have access yet — GitHub hasn’t enabled model switching on your account.

You can confirm this by checking:

https://github.com/features/copilot

Or your subscription settings at: https://github.com/settings/copilot

🛠️ Summary:
You have the correct versions ✅

The dropdown may be missing because:

You're not using the correct chat panel

Or GitHub hasn’t enabled model selection for your account yet 🚫

**Comment by ArshAliAzam** [comment]
You're looking at the GitHub Copilot settings, but the model selection dropdown (used to pick between different AI models like GPT-3.5 or GPT-4) is not showing.

✅ Here's what to check:
1. Copilot Labs vs Copilot Chat
Make sure you are using GitHub Copilot Chat, not just Copilot Labs.
The model dropdown appears only in Copilot Chat (inside the chat panel, near the top).

2. Use the “Change Copilot Completion Model” command
Try this:

Press Ctrl + Shift + P (or Cmd + Shift + P on Mac) to open the Command Palette.

Type:
"Change Copilot Completion Model"

  If this option shows up, click it.

  You should then see the list of available models (like gpt-3.5-turbo, gpt-4, etc.)

  If the command doesn't appear, it means you don’t have access to model selection yet (see point 4 below).

3. Check if you're signed in correctly
Make sure:

  You're signed into GitHub inside VS Code.

  You have an active Copilot subscription (especially Copilot Chat Beta).

4. Model selection is in limited rollout
GitHub is gradually rolling out model selection. If you don't see the dropdown or command, it likely means:

  ❌ You don't have access yet — GitHub hasn’t enabled model switching on your account.

  You can confirm this by checking:

  https://github.com/features/copilot

  Or your subscription settings at: https://github.com/settings/copilot

🛠️ Summary:
You have the correct versions ✅

The dropdown may be missing because:

You're not using the correct chat panel

Or GitHub hasn’t enabled model selection for your account yet 🚫

**Comment by zatricky** [comment]
I found I had this same issue where I had the Modes button (Agent/Ask/Edit) but not a Model button upon starting up vscode after updating to 1.103 today. The option to switch models appeared only after I submitted a prompt, at which point it showed I was still using the previous model I was using the day before. And of course I am now able to switch models. Hopefully this helps someone. :-)

As to the cause, this option is unavailable when you are not logged in. My suspicion is that the option is not showing after login because something in the UI has not yet updated - and submitting a prompt forces this to refresh in some way.

**Reply by ovidb** [reply]
Same here, wondering if we should be starting a new conversation

**Reply by kamil-kubiczek** [reply]
Same for me

**Comment by bovel84** [comment]
Title:
Can’t select agents in GitHub Copilot for VS Code anymore

Body:
Hi,
Until yesterday I was able to select different agents in GitHub Copilot for VS Code. Now, the agents section has disappeared and I can only see “Ask agent” and “Edit”.
I haven’t changed any settings or updated anything manually.
Is this a known issue? Are other users experiencing the same problem?
Thanks for your help

Reproduction Steps

In vs code not work agent

Fault Type

Permanent - Occurring repeatably

Blocking?

Yes

Last working date

14 August

Logs

**Comment by BurtHarris** [comment]
I'm having the same problem, Works on github website, but not in vscode.   

## Environment

- Plugin: GitHub Copilot for Visual Studio Code 1.368.0
- Editor: Visual Studio Code 1.104.0
- Operating System: Windows_NT 10.0.26100 (x64)

## Copilot

- Version: 1.368.0
- GitHub Account: BurtHarris
- Session ID: d7ff6f9a-a51f-4dac-bf95-4b4d3e4a13e51757648007549
- Send Restricted Telemetry: enabled
- Content Exclusion: unavailable

## Network Configuration

- Proxy: N/A
- Proxy Authorization: N/A
- Proxy Kerberos SPN: N/A
- Proxy Strict SSL: enabled
- Fetcher: Helix
- Number of Root Certificates: 150
- TLS Default Min Version: TLSv1.2
- TLS Default Max Version: TLSv1.3

## Reachability

- github.com: HTTP 200
- api.github.com: HTTP 200
- proxy.individual.githubcopilot.com: HTTP 200
- api.individual.githubcopilot.com: HTTP 200
- telemetry.individual.githubcopilot.com: HTTP 200

# Screenshot
<img width="1921" height="1080" alt="copilot model selector missing" src="https://github.com/user-attachments/assets/6960dc3c-91d6-4c24-a306-142e3d29be17" />

# Lengthy chat with/Copilot attempting to resolve the problem

https://github.com/copilot/share/0072420e-40c0-84e5-9952-0c03c4c7605f

**Reply by drgrib** [reply]
Started having the same problem today in macOS. But then the option reappeared after a couple minutes.

**Comment by BurtHarris** [comment]
I tested it on the VSCode Insiders edition; it seems to work there.
